### PR TITLE
_getValidity is supposed to take an argument

### PR DESCRIPTION
**Pull request by rictic**
The compiler pointed out that `IronValidatableBehavior` wants `_getValidity` to be told what value to validate, rather than to introspect for itself.

Also, use `!!` to coerce to boolean.

### CODE REVIEW

**Comment by rictic** [comment]
Hm, looks like this breaks tests. I'll take a longer look into this.

**Comment by notwaldorf** [comment]
Yeahhh I don't think that's going to work. `getValidity` is called from IronValidatableBehavior's `validate()` method, and I am not sure it always has a `value` passed to it (hence the `this.value`). What happens if you add the argument to make the compiler happy but don't use it?

**Comment by rictic** [comment]
That's what I've done internally, and it has appeased the compiler.

On the face of it though, it does kinda seem like either `_getValidity` should always take an argument or it never should. WDYT?

**Comment by notwaldorf** [comment]
Ugh the problem is that we're using IronValidatableBehaviour kinda poorly (and this is mostly my fault). It has a `validate(value)` method, that sometimes we use with an argument (in `PaperInputContainer`, when it's trying to auto-validate its contents, https://github.com/PolymerElements/paper-input/blob/master/paper-input-container.html#L495), and sometimes we use without (like, say, when in a form you would do `element.validate()`). 

I'd like to clean this up because it bugs me, but I can't quite right now (I'd like to clean up the gold elements first, so that they don't get all broken by whatever changes we make to paper-input). Also, something something breaking changes.

TL; DR: :sob:

**Comment by danbeam** [comment]
can we just document that the `@param` is optional or could be `undefined`?

**Comment by rictic** [comment]
Optional seems reasonablish to me, to express what we're currently doing

**Comment by rictic** [comment]
I'll take a shot at it.

**Comment by rictic** [comment]
Replaced by #107
